In [2]:
import pandas as pd

In [3]:
data = pd.read_csv('data_cleaned.csv', delimiter=",")
data = data.drop(columns=["Unnamed: 0", "Q25_First Click"])

In [4]:
data.head(34)

Duration (in seconds) UserLanguage  participant_number  age  gender  \
0                    1837           FR                  26   22       2   
1                    2522           FR                  59   24       2   
2                    2356           EN                 136   27       2   
3                    1168           EN                  55   23       2   
4                    1291           EN                 106   22       1   
5                    1157           FR                  25   23       2   
6                    2915           EN                  52   21       2   
7                    3043           FR                  57   23       2   
8                    5843           FR                 137   21       2   
9                    1030           FR                 109   22       2   
10                   1023           FR                 129   20       2   
11                   3102           FR                  67   28       1   
12                   3979           FR                  60   24       2   
13                   4629           FR                  42   23       1   
14                   2989           FR                 110   31       1   
15                   2312           EN                 120   23       2   
16                   3272           FR                  14   23       2   
17                   1362           EN                  69   25       1   
18                   1055           FR                  65   22       1   
19                   1522           FR                  88   19       1   
20                   4172           FR                 127   26       1   
21                   5291           FR                  19   22       2   
22                   3593           FR                  85   24       1   
23                   2820           FR                  24   19       2   
24                   3299           FR                 135   24       2   
25                   3405           FR                   6   23       1   
26                   1242           FR                  31   23       2   
27                   1061           FR                  73   21       2   
28                   1301           FR                 133   22       2   
29                   2390           EN                  83   23       1   
30                   2426           FR                 138   33       2   
31                    963           FR                  30   23       2   
32                   1342           FR                  89   21       1   
33                   1431           FR                 107   23       2   

    ethnicity  political_orientation  \
0           3                      3   
1           2                      3   
2           1                      1   
3           2                      1   
4           1                      1   
5           3                      2   
6           1                      1   
7           1                      1   
8           1                      2   
9           1                      2   
10          1                      2   
11          1                      1   
12          1                      2   
13          1                      4   
14          1                      1   
15          1                      2   
16          1                      2   
17          1                      1   
18          1                      3   
19          1                      4   
20          1                      4   
21          1                      1   
22          3                      2   
23          1                      1   
24          1                      3   
25          1                      1   
26          1                      4   
27          1                      2   
28          1                      2   
29          1                      1   
30          1                      4   
31          1                      1   
32          2                      1   
33          1     

In [5]:
# remove veggie and vegan participants
index_veg = data.index[(data.diet == 1) | (data.diet == 2)]
excluded_participants = data.loc[index_veg]
excluded_participants["reason_to_exclude"] = "Veggie/Vegan"
data = data.drop(index_veg)

In [6]:
# remove participants who didn't take any sandwich
index_noSandwich = data.index[data.sandwich == 5]
excluded_participants = pd.concat([excluded_participants, data.loc[index_noSandwich]])
excluded_participants["reason_to_exclude"] = excluded_participants["reason_to_exclude"].fillna("No sandwich taken")
data = data.drop(index_noSandwich)

In [7]:
# Have a look at other diets
not_omni = data.drop(data.index[data.diet == 3])
not_omni.head(34) # we have 5 people, all said "other"

Duration (in seconds) UserLanguage  participant_number  age  gender  \
3                    1168           EN                  55   23       2   
10                   1023           FR                 129   20       2   
12                   3979           FR                  60   24       2   
14                   2989           FR                 110   31       1   
15                   2312           EN                 120   23       2   

    ethnicity  political_orientation political_orientation_text  \
3           2                      1                        NaN   
10          1                      2                        NaN   
12          1                      2                        NaN   
14          1                      1                        NaN   
15          1                      2                        NaN   

    achieved_education achieved_education_text  ...  bothered_post  \
3                    4                     NaN  ...              5   
10                   3                     NaN  ...              0   
12                   3                     NaN  ...              0   
14                   4                     NaN  ...              0   
15                   3                     NaN  ...              0   

    uneasy_post  uncomfortable_post  sandwich  diet  \
3             0                   0         2     7   
10            0                   0         2     7   
12            0                   7         3     7   
14            0                   0         4     7   
15            0                   0         1     7   

                                            diet_text  \
3                         No red meat (pork included)   
10                                   pesco vegetarien   
12                          Pas de régime particulier   
14                                       Fléxitarien    
15  Flexitarian; Vegetarian and Dairy free majorit...   

                                    reason_for_choice  labels_attention  \
3                  It was vegetarian and looked tasty                 2   
10  me donnait le plus envie et je ne mange pas de...                 1   
12  J'hésitais entre jambon/fromage et viande séch...                 1   
14      Rare que je puisse manger de la viande séché                  1   
15  Because it looked the most interesting; fit wi...                 1   

    feeling_observed  did_value_activation_affect_choice  
3                  2                                 NaN  
10                 2                                 2.0  
12                 2                                 NaN  
14                 2                                 2.0  
15                 2                                 2.0  

[5 rows x 29 columns]

### Diets of participants that said "other"
55: no read meat
129: pesco-vegetarian
60: no particular diet
110: flexitarian (meat sandwich): Dr. Psycho, he told us his rule was only eat meat outside the house
120: Flexitarian: Vegetarian and Dairy free majority of the time; but when travelling or a guest am flexible with diet
-> I would say we keep 110 since his rule did allow him to eat meat and exclude 120 since their definition of flexitarian probably means that they would not eat meat in this situation

In [8]:
# remove participant 55, 129, 120
index_otherDiet = data.index[(data.participant_number == 55) | (data.participant_number == 129) | (data.participant_number == 120)]
excluded_participants = pd.concat([excluded_participants, data.loc[index_otherDiet]])
excluded_participants["reason_to_exclude"] = excluded_participants["reason_to_exclude"].fillna("Other diets similar to veggie")
data = data.drop(index_otherDiet)

In [9]:
# we remove participant 88 because he took the break at the wrong time
index_wrongBreak = data.index[data.participant_number == 88]
excluded_participants = pd.concat([excluded_participants, data.loc[index_wrongBreak]])
excluded_participants["reason_to_exclude"] = excluded_participants["reason_to_exclude"].fillna("Took the break at the wrong time")
data = data.drop(index_wrongBreak)

In [10]:
excluded_participants = excluded_participants[["participant_number", "reason_to_exclude"]]

In [11]:
excluded_participants.head(20)

participant_number                 reason_to_exclude
2                  136                      Veggie/Vegan
7                   57                      Veggie/Vegan
11                  67                      Veggie/Vegan
23                  24                      Veggie/Vegan
32                  89                      Veggie/Vegan
0                   26                 No sandwich taken
5                   25                 No sandwich taken
13                  42                 No sandwich taken
16                  14                 No sandwich taken
30                 138                 No sandwich taken
31                  30                 No sandwich taken
33                 107                 No sandwich taken
3                   55     Other diets similar to veggie
10                 129     Other diets similar to veggie
15                 120     Other diets similar to veggie
19                  88  Took the break at the wrong time

In [12]:
data.head(40)

Duration (in seconds) UserLanguage  participant_number  age  gender  \
1                    2522           FR                  59   24       2   
4                    1291           EN                 106   22       1   
6                    2915           EN                  52   21       2   
8                    5843           FR                 137   21       2   
9                    1030           FR                 109   22       2   
12                   3979           FR                  60   24       2   
14                   2989           FR                 110   31       1   
17                   1362           EN                  69   25       1   
18                   1055           FR                  65   22       1   
20                   4172           FR                 127   26       1   
21                   5291           FR                  19   22       2   
22                   3593           FR                  85   24       1   
24                   3299           FR                 135   24       2   
25                   3405           FR                   6   23       1   
26                   1242           FR                  31   23       2   
27                   1061           FR                  73   21       2   
28                   1301           FR                 133   22       2   
29                   2390           EN                  83   23       1   

    ethnicity  political_orientation  political_orientation_text  \
1           2                      3                         NaN   
4           1                      1                         NaN   
6           1                      1                         NaN   
8           1                      2                         NaN   
9           1                      2                         NaN   
12          1                      2                         NaN   
14          1                      1                         NaN   
17          1                      1                         NaN   
18          1                      3                         NaN   
20          1                      4                     absente   
21          1                      1                         NaN   
22          3                      2                         NaN   
24          1                      3                         NaN   
25          1                      1                         NaN   
26          1                      4  aucune orientation précise   
27          1                      2                         NaN   
28          1                      2                         NaN   
29          1                      1                         NaN   

    achieved_education achieved_education_text  ...  bothered_post  \
1                    3                     NaN  ...              0   
4                    3                     NaN  ...             90   
6                    2                     NaN  ...              0   
8                    3                     NaN  ...              3   
9                    3                     NaN  ...              0   
12                   3                     NaN  ...              0   
14                   4                     NaN  ...              0   
17                   3                     NaN  ...              9   
18                   2                     NaN  ...              0   
20                   4                     NaN  ...             66   
21                   3                     NaN  ...              0   
22                   2                     NaN  ...             12   
24                   3                     NaN  ...              1   
25                   3                     NaN  ...              0   
26                   3                     NaN  ...              0   
27                   8     collégiale (quebec)  ...              0   
28                   3                     NaN  ...             10   
29                   3   

In [13]:
data.to_csv('data_final_participants.csv', sep = ',')
excluded_participants.to_csv('excluded_participants.csv', sep = ',')